execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1635,17 +1637,26 @@ where
             .map_err(Error::from)
     }
 
-    pub fn put_trie(
+    pub fn put_trie_and_find_missing_descendant_trie_keys(
         &self,
         correlation_id: CorrelationId,
         trie: &Trie<Key, StoredValue>,
-    ) -> Result<(), Error>
+    ) -> Result<InsertedTrieKeyAndMissingDescendants, Error>
     where
         Error: From<S::Error>,
     {
-        self.state
+        let inserted_trie_key = self
+            .state
             .put_trie(correlation_id, trie)
-            .map_err(Error::from)
+            .map_err(Error::from)?;
+        let missing_descendant_trie_keys = self
+            .state
+            .missing_trie_keys(correlation_id, inserted_trie_key)
+            .map_err(Error::from)?;
```

Comment:
Should it even be possible to insert a trie with missing descendants? Or should that wait in some queue until we have all descendants?

Reply:
> Should it even be possible to insert a trie with missing descendants?

The `put_trie` methods makes this possible yeah.

> Or should that wait in some queue until we have all descendants?

The idea is to have a queue.

[This unit test in the EE][1] demonstrates the loop for how fast sync is going to use.

[1]: http://github.com/xcthulhu/casper-node/blob/master/execution_engine/src/storage/trie_store/operations/tests/synchronize.rs#L52-L80